### ISSUE TITLE
[REVIEW]: fastPLI: A Fiber Architecture Simulation Toolbox for 3D-PLI

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3042]
**Submitting author:** @fmatuschke (<a href="http://orcid.org/0000-0001-6435-5351">Felix Matuschke</a>)
**Repository:** <a href="https://github.com/3d-pli/fastpli" target ="_blank">https://github.com/3d-pli/fastpli</a>
**Version:** 1.1.0
**Editor:** @oliviaguest
**Reviewers:** @vigji, @glyg, @RealPolitiX
**Archive:** <a href="https://doi.org/10.5281/zenodo.4720075" target="_blank">10.5281/zenodo.4720075</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/0c61ea3e028958db084f788747481dab/status.svg)](https://joss.theoj.org/papers/0c61ea3e028958db084f788747481dab)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/0c61ea3e028958db084f788747481dab"><img src="https://joss.theoj.org/papers/0c61ea3e028958db084f788747481dab/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/0c61ea3e028958db084f788747481dab/status.svg)](https://joss.theoj.org/papers/0c61ea3e028958db084f788747481dab)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/3d-pli/fastpli" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@vigji, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @oliviaguest know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/3d-pli/fastpli">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@fmatuschke) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/3d-pli/fastpli">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@fmatuschke) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/3d-pli/fastpli">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@fmatuschke) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @vigji it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3389/fninf.2011.00034 is OK
- 10.3233/APC190017 is OK
- 10.1103/physrevx.10.021002 is OK
- 10.17815/jlsrf-2-121 is OK
- 10.3389/fnana.2018.00075 is OK
- 10.1016/j.neuroimage.2015.02.020 is OK
- 10.1098/rsif.2015.0734 is OK
- 10.1145/2833157.2833162 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.34 s (347.2 files/s, 110372.8 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
SVG                              4              2              2          21804
Python                          49           1601           1480           4518
C++                             15            692            202           2978
C/C++ Header                    19            304             69           1451
Markdown                         4            115              0            275
CMake                            5             39             36            201
YAML                             3             16             21            180
Jupyter Notebook                 2              0            418            176
Bourne Shell                    10             30             14            165
make                             2             34             10            145
TeX                              1              8              0            100
reStructuredText                 3             17             80              9
-------------------------------------------------------------------------------
SUM:                           117           2858           2332          32002
-------------------------------------------------------------------------------


Statistical information for the repository '845d1dee56dd4b12c318a43e' was
gathered on 2021/02/15.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
fmatuschke                     658         30396          17112          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
fmatuschke                13295           43.7         11.1                6.30
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03042/joss.03042/10.21105.joss.03042.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03042/joss.03042/10.21105.joss.03042.pdf) :page_facing_up: :point_left:

**Comment by oliviaguest** [comment]
Thank you so much, @vigji for accepting to review this. I want to find at least one more reviewer, but since you started reviewing already, please feel free to indeed get started.

Please read the instructions above. If you have trouble ticking things off in the list above, remember you need to click to "accept" here: https://github.com/openjournals/joss-reviews/invitations! Any questions, feedback on the paper, etc., please post here. Any very code-specific questions, suggestions, etc., please use the issues in the code repo and link to them from this thread, so we can all keep track of them. 🌸

For examples of how this process plays out feel free to skim previous reviews, such as: #2285 and #2348. ☺️

**Comment by oliviaguest** [comment]
@whedon invite @glyg as reviewer

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by oliviaguest** [comment]
@whedon add @glyg as reviewer

**Comment by whedon** [comment]
OK, @glyg is now a reviewer

**Comment by oliviaguest** [comment]
👋  @glyg welcome! Make sure to read the instructions above, accept the invitation, etc.

**Comment by vigji** [comment]
I'll link here below all issues related to this review:

https://github.com/3d-pli/fastpli/issues/1

https://github.com/3d-pli/fastpli/issues/2

https://github.com/3d-pli/fastpli/issues/3

https://github.com/3d-pli/fastpli/issues/4

https://github.com/3d-pli/fastpli/issues/5

https://github.com/3d-pli/fastpli/issues/6

**Comment by oliviaguest** [comment]
@whedon add @RealPolitiX as reviewer

**Comment by whedon** [comment]
OK, @RealPolitiX is now a reviewer

**Comment by oliviaguest** [comment]
Oooh, and also: 👋 @RealPolitiX welcome! Make sure to read the instructions above, accept the invitation, etc.

Thank you all again!

**Comment by oliviaguest** [comment]
Hey @vigji, @glyg, @RealPolitiX, when you all get a chance can you post your ETAs for your reviews — thanks! 🌸

**Comment by glyg** [comment]
Hi @oliviaguest I plan to make some progress on that at the end of the week, not sure how far I'll be able to get, but I'll update on Friday with an ETA

**Comment by vigji** [comment]
Hey @oliviaguest ! I did most of my comments on the package via issues in the repo, will write a complete report once they will all be addressed - If I got the procedure right!

**Comment by glyg** [comment]
Hi all

First, congrats @fmatuschke on the software, it is a really nice piece of work.  I opened several issues on fastPLI repo for technical concerns.

Here are aspects more related to the paper itself:


### State of the field:  Do the authors describe how this software compares to other commonly-used packages?

Here I failed to see a review of other similar software or efforts. The cited literature focuses on the authors' team. Although I understand the imaging modality is specific, is it not used in other settings? Simulation of IRM data is alluded to, maybe this adjacent field has more references? As is it is hard to situate this work in the general field of brain/neuron imagery.
A more scholarly (minor) question: I know 2 photons / second harmonics generation methods are also used to image bi-refringent tissues. Could fastPLI be adapted to modeling this modality? Are there comparable modeling efforts for  these techniques?

### Quality of writing: Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?

Some revision of the writing should be made. @oliviaguest should I provide a detailed list of typos and such? 

### References: Is the list of references complete, and is everything cited appropriately that should be cited

As stated in the first section, a review of the field or references to contributions from outside the authors' team (unless I missed something) would be welcome.

### Opened issues:

I opened 4 issues 

The first 2 require minor efforts [#7](https://github.com/3d-pli/fastpli/issues/7), [#8](https://github.com/3d-pli/fastpli/issues/8) and are cosmetic.

The next 2 ([#9](https://github.com/3d-pli/fastpli/issues/9), [#10](https://github.com/3d-pli/fastpli/issues/10) ) would improve the software distribution and visualization, but I don't consider they need to be solved prior to publication.



In summary, I think this paper should be published  pending the minor revisions in paper.md outlined above.

**Comment by whedon** [comment]
:wave: @vigji, please update us on how your review is going (this is an automated reminder).

**Comment by oliviaguest** [comment]
@vigji I'd prefer if you post them all here now so I (and others) can see and comment too, if need be. Thanks!

**Comment by fmatuschke** [comment]
@glyg thanks for the compliment. Since this is my first official attempt at releasing software, it means a lot.
All the issues you opened in the software repository were commented by me. They are nice suggestions for improvements.
As for the missing discussion of other software, I definitely failed to address that. I will add this as soon as possible and let you all know.

**Comment by oliviaguest** [comment]
@whedon re-invite @RealPolitiX as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@realpolitix please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by RealPolitiX** [comment]
The paper from @fmatuschke is well-written apart from a few grammar errors. The references are thorough. I think due to the novelty of the measurement technique used for reconstruction of neuronal fibers, it's not used elsewhere in general. You can probably cite some of the references from [OpenPolScope](https://openpolscope.org/pages/Literature.htm), since it's related and discuss your unique context.

As for the software, I opened [issue #11](https://github.com/3d-pli/fastpli/issues/11). The wiki looks nicely written.

**Comment by fmatuschke** [comment]
I added the missing section about the software compares mentioned by @glyg.
Additionally I fixed some typos I found.

Thank you @RealPolitiX for opening https://github.com/3d-pli/fastpli/issues/11. I fixed it.

Just a small summary for @all.
I have addressed all the issues so far. The main changes since the opening of the review are a complete rework of my Wiki and API documentation.

**Comment by oliviaguest** [comment]
@fmatuschke thanks for the update. Can you please all —  @vigji, @glyg, @RealPolitiX — make sure everything that has not yet been ticked off is and/or discuss any (new) changes you might be looking for? Ta! ☺️

**Comment by glyg** [comment]
Looks good to me now :smile:

**Comment by oliviaguest** [comment]
I assume the other two reviewers are not checking their notifications, I will email them now.

**Comment by vigji** [comment]
I am very sorry, yes, I was missing the notifications!

I went again through the repository. For the review, I opened 6 issues (https://github.com/3d-pli/fastpli/issues/1, https://github.com/3d-pli/fastpli/issues/2, https://github.com/3d-pli/fastpli/issues/3, https://github.com/3d-pli/fastpli/issues/4, https://github.com/3d-pli/fastpli/issues/5, https://github.com/3d-pli/fastpli/issues/6) concerning installation, automatic testing, clarity of the documentation, and internal organisation and naming consistency in the package. All of them have been thoroughly addressed by the author. Kudos to the new wiki, it is extensive and very readable now! I could successfully install the package and ran the tests.

I checked the last version of the manuscript and it looks good to me. I don't have the expertise in the subject of interest to suggest other related packages and publications that should be discussed in the paper,  but the literature now discussed in the context of fibres models seems good to me!

All my issues have been addressed and I have no further requirements for the paper to be published. Nice job @fmatuschke !

**Comment by oliviaguest** [comment]
@fmatuschke when you feel everything has been decided to be "ready", let know (get whedon to compile the most recent PDF, etc.)... and I'll pop back in here and move this along. ☺️

**Comment by fmatuschke** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03042/joss.03042/10.21105.joss.03042.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03042/joss.03042/10.21105.joss.03042.pdf) :page_facing_up: :point_left:

**Comment by oliviaguest** [comment]
@fmatuschke is it looking good? ☺️  If so, can you deposit the code in zenodo or similar and post the DOI back here?

**Comment by fmatuschke** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03042/joss.03042/10.21105.joss.03042.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03042/joss.03042/10.21105.joss.03042.pdf) :page_facing_up: :point_left:

**Comment by fmatuschke** [comment]
@all I fixed hopefully the last typo and one reference in the paper.
@oliviaguest I published the software in zenodo with the DOI: [10.5281/zenodo.4720075](https://zenodo.org/record/4720075)

**Comment by oliviaguest** [comment]
@whedon set 10.5281/zenodo.4720075 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4720075" target="_blank">10.5281/zenodo.4720075</a> is the archive.

**Comment by oliviaguest** [comment]
@whedon check references

**Comment by oliviaguest** [comment]
@whedon check repository

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.50 s (227.7 files/s, 75484.8 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
SVG                              4              2              2          21804
Python                          46           1506           1724           4006
C++                             15            692            199           2978
C/C++ Header                    19            304             69           1451
Jupyter Notebook                 4              0            794            369
Markdown                         4            129              0            312
CMake                            5             42             38            208
TeX                              1             15              0            180
make                             2             34             10            159
YAML                             3             10             19            147
Bourne Shell                     7             25             10            116
reStructuredText                 3             18             85             11
-------------------------------------------------------------------------------
SUM:                           113           2777           2950          31741
-------------------------------------------------------------------------------


Statistical information for the repository 'ad085f14f1572dcad7d76238' was
gathered on 2021/04/26.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
fmatuschke                     680         31099          18181          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
fmatuschke                12929           41.6         13.1                6.19
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3389/fninf.2011.00034 is OK
- 10.3233/APC190017 is OK
- 10.1103/physrevx.10.021002 is OK
- 10.17815/jlsrf-4-121-1 is OK
- 10.3389/fnana.2018.00075 is OK
- 10.1016/j.neuroimage.2015.02.020 is OK
- 10.1098/rsif.2015.0734 is OK
- 10.1145/2833157.2833162 is OK
- 10.1103/physreve.83.041804 is OK
- 10.3389/fninf.2017.00005 is OK
- 10.1016/j.neuroimage.2019.02.055 is OK
- 10.1364/opex.13.004420 is OK
- 10.1117/1.3241986 is OK
- 10.1016/j.optcom.2020.126113 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by oliviaguest** [comment]
@whedon generate pdf

**Comment by oliviaguest** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3389/fninf.2011.00034 is OK
- 10.3233/APC190017 is OK
- 10.1103/physrevx.10.021002 is OK
- 10.17815/jlsrf-4-121-1 is OK
- 10.3389/fnana.2018.00075 is OK
- 10.1016/j.neuroimage.2015.02.020 is OK
- 10.1098/rsif.2015.0734 is OK
- 10.1145/2833157.2833162 is OK
- 10.1103/physreve.83.041804 is OK
- 10.3389/fninf.2017.00005 is OK
- 10.1016/j.neuroimage.2019.02.055 is OK
- 10.1364/opex.13.004420 is OK
- 10.1117/1.3241986 is OK
- 10.1016/j.optcom.2020.126113 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03042/joss.03042/10.21105.joss.03042.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03042/joss.03042/10.21105.joss.03042.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2262

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2262, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@fmatuschke I have proofread your paper and have some very minor comments which I hope you can address:

- [ ] Please check `It allows contrasting fibers and fiber tracts and ultimately reconstructing 3D nerve fiber orientations`, some words seem missing. Should this be something like `It provides image contrast for fibers and fiber tracts, and ultimately enables reconstruction of 3D nerve fiber orientations`?
- [ ] Please introduce the acronym dMRI for diffusion magnetic resonance imaging, when the terms are first used/introduced. 
- [ ] Consider changing `Different types of simulation...` to `Different types of simulations...`

Once you have considered the above please call `@whedon generate pdf` to update the paper. I will then review the changes and process acceptance when ready. 

One more important note relates to the Zenodo archive
- [ ] Please ensure that the meta-data for the Zenodo archive matches that of the paper in terms of the author list as well as the title. Currently the archive title read: `Fiber Architecture Simulation Toolbox for 3D-PLI` but this should match the paper and be changed to `fastPLI: A Fiber Architecture Simulation Toolbox for3D-PLI`. You can manually change this on Zenodo. 

The version tag for the Zenodo archive is `1.1.0`. I will therefore assume we should update the version tag here also. I will do this shortly below :point_down:. 

Thanks.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set 1.1.0 as version

**Comment by whedon** [comment]
OK. 1.1.0 is the version.

**Comment by fmatuschke** [comment]
Dear @Kevin-Mattheus-Moerman,
I change all three of your comments. Thanks for your help.
Regarding the version number. Yes, it is now officially 1.1. Thanks to the reviewers I changed quite a bit, so I had to increase the minor version.
I have also corrected the title in the Zenodo archive.

**Comment by fmatuschke** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03042/joss.03042/10.21105.joss.03042.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03042/joss.03042/10.21105.joss.03042.pdf) :page_facing_up: :point_left:

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set version as 1.1

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set version as 1.1.0

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3389/fninf.2011.00034 is OK
- 10.3233/APC190017 is OK
- 10.1103/physrevx.10.021002 is OK
- 10.17815/jlsrf-4-121-1 is OK
- 10.3389/fnana.2018.00075 is OK
- 10.1016/j.neuroimage.2015.02.020 is OK
- 10.1098/rsif.2015.0734 is OK
- 10.1145/2833157.2833162 is OK
- 10.1103/physreve.83.041804 is OK
- 10.3389/fninf.2017.00005 is OK
- 10.1016/j.neuroimage.2019.02.055 is OK
- 10.1364/opex.13.004420 is OK
- 10.1117/1.3241986 is OK
- 10.1016/j.optcom.2020.126113 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2274

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2274, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1389139217662500865) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2275
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03042](https://doi.org/10.21105/joss.03042)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by Kevin-Mattheus-Moerman** [comment]
Congratulations @fmatuschke this work is now published in JOSS. 
Thank you @oliviaguest for editing this submission! 

Thanks also to @vigji, @glyg, @RealPolitiX for your help reviewing this work!!!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03042/status.svg)](https://doi.org/10.21105/joss.03042)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03042">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03042/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03042/status.svg
   :target: https://doi.org/10.21105/joss.03042
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03042/status.svg)](https://doi.org/10.21105/joss.03042)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
